### ISSUE TITLE
Fail to compile on Mac

### ISSUE THREAD

**Issue by moh-abk** [fede1024/rust-rdkafka#135]
Looking to use this crate and get the following error when doing a `cargo run`

`cargo.toml` -- `rdkafka = "~0.21"`

```
   Compiling rdkafka-sys v1.0.0
error: failed to run custom build command for `rdkafka-sys v1.0.0`

Caused by:
  process didn't exit successfully: `/Users/xxx/xxx/xxx/xxx/target/debug/build/rdkafka-sys-d6a1b4047ca8705e/build-script-build` (exit code: 101)
--- stdout
Configuring librdkafka
using cache file config.cache
checking for OS or distribution... ok (osx)
checking for C compiler from CC env... failed
checking for gcc (by command)... ok (cached)
checking for C++ compiler from CXX env... failed
checking for C++ compiler (g++)... ok (cached)
checking executable ld... ok (cached)
checking executable nm... ok (cached)
checking executable objdump... ok (cached)
checking executable strip... ok (cached)
checking for debug symbols compiler flag (-g...)... ok (cached)
checking for pkgconfig (by command)... ok (cached)
checking for install (by command)... failed
checking for PIC (by compile)... ok (cached)
checking for GNU-compatible linker options... failed
checking for OSX linker options... ok (cached)
checking for GNU linker-script ld flag... failed
checking for Solaris linker-script ld flag... failed (ignore)
checking for __atomic_32 (by compile)... ok (cached)
checking for __atomic_64 (by compile)... ok (cached)
checking for socket (by compile)... ok (cached)
parsing version '0x010000ff'... ok (1.0.0)
checking for librt (by pkg-config)... failed
checking for librt (by compile)... failed
checking for libpthread (by pkg-config)... failed
checking for libpthread (by compile)... ok (cached)
checking for c11threads (by pkg-config)... failed
checking for c11threads (by compile)... failed (disable)
checking for libdl (by pkg-config)... failed
checking for libdl (by compile)... ok (cached)
checking for zlib (by pkg-config)... ok
checking for zlib (by compile)... ok (cached)
checking for libcrypto (by pkg-config)... ok
checking for libcrypto (by compile)... ok (cached)
checking for libssl (by pkg-config)... ok
checking for libssl (by compile)... ok (cached)
checking for libsasl2 (by pkg-config)... failed
checking for libsasl2 (by compile)... ok (cached)
checking for zstd (by pkg-config)... failed
checking for zstd (by compile)... ok (cached)
checking for libm (by pkg-config)... failed
checking for libm (by compile)... ok (cached)
checking for liblz4 (by pkg-config)... ok
checking for liblz4 (by compile)... ok (cached)
checking for rapidjson (by compile)... failed (disable)
checking for crc32chw (by compile)... ok (cached)
checking for regex (by compile)... ok (cached)
checking for strndup (by compile)... ok (cached)
checking for strerror_r (by compile)... ok (cached)
checking for pthread_setname_gnu (by compile)... failed (disable)
checking for nm (by env NM)... ok (cached)
Generated Makefile.config
Generated config.h

Configuration summary:
  prefix                   /usr/local
  MKL_DISTRO               osx
  SOLIB_EXT                .dylib
  ARCH                     x86_64
  CPU                      generic
  GEN_PKG_CONFIG           y
  ENABLE_SSL               y
  ENABLE_GSSAPI            y
  ENABLE_DEVEL             n
  ENABLE_VALGRIND          n
  ENABLE_REFCNT_DEBUG      n
  ENABLE_SHAREDPTR_DEBUG   n
  ENABLE_LZ4_EXT           y
  MKL_APP_NAME             librdkafka
  MKL_APP_DESC_ONELINE     The Apache Kafka C/C++ library
  CC                       gcc
  CXX                      g++
  LD                       ld
  NM                       nm
  OBJDUMP                  objdump
  STRIP                    strip
  CFLAGS                   -I/usr/local/Cellar/openssl/1.0.2s/include  -I/usr/local/Cellar/openssl/1.0.2s/include -I/usr/local/Cellar/openssl/1.0.2s/include -I/usr/local/Cellar/lz4/1.9.1/include
  CPPFLAGS                 -gstrict-dwarf -O2 -fPIC -Wall -Wsign-compare -Wfloat-equal -Wpointer-arith -Wcast-align
  PKG_CONFIG               pkg-config
  INSTALL                  install
  LIB_LDFLAGS              -shared -dynamiclib -Wl,-install_name,$(DESTDIR)$(libdir)/$(LIBFILENAME)
  RDKAFKA_VERSION_STR      1.0.0
  MKL_APP_VERSION          1.0.0
  LIBS                     -L/usr/local/Cellar/lz4/1.9.1/lib -llz4 -lm -lzstd -lsasl2 -L/usr/local/Cellar/openssl/1.0.2s/lib -lssl -L/usr/local/Cellar/openssl/1.0.2s/lib -lcrypto -lz -ldl -lpthread
  CXXFLAGS                 -Wno-non-virtual-dtor
  SYMDUMPER                $(NM) -g
  exec_prefix              /usr/local
  bindir                   /usr/local/bin
  sbindir                  /usr/local/sbin
  libexecdir               /usr/local/libexec
  datadir                  /usr/local/share
  sysconfdir               /usr/local/etc
  sharedstatedir           /usr/local/com
  localstatedir            /usr/local/var
  libdir                   /usr/local/lib
  includedir               /usr/local/include
  infodir                  /usr/local/info
  mandir                   /usr/local/man
  BUILT_WITH               GCC GXX PKGCONFIG OSXLD LIBDL PLUGINS ZLIB SSL SASL_CYRUS ZSTD HDRHISTOGRAM LZ4_EXT SNAPPY SOCKEM SASL_SCRAM CRC32C_HW
Generated config.cache

Now type 'make' to build

--- stderr
Building and linking librdkafka statically
Running command: "./configure " in dir: librdkafka
thread 'main' panicked at 'Command failed with error: No such file or directory (os error 2)', /Users/xxxx/.cargo/registry/src/github.com-1ecc6299db9ec823/rdkafka-sys-1.0.0/build.rs:25:21
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace.
```

**Comment by moh-abk** [comment]
digging into https://github.com/fede1024/rust-rdkafka/blob/master/rdkafka-sys/build.rs it `cargo run` works correctly when i set the `CARGO_FEATURE_DYNAMIC_LINKING ` env var

**Comment by dreamersdw** [comment]
The compilation will be success when changing `rdkafka-sys-1.0.0/build.sh` line 109 from
```rust
run_command_or_fail("librdkafka", "./configure", configure_flags.as_slice());
```
to
```rust
run_command_or_fail("librdkafka", "<full path of ./configure>", configure_flags.as_slice());
```
I'll try to find the real reason behind it a while later.

**Comment by benesch** [comment]
Fixed in #161.
